canister/src/blocktree.rs

```diff
@@ -185,19 +185,14 @@ fn get_chain_with_tip_reverse<'a, 'b>(
     None
 }
 
-/// Returns the depth of the tree.
-pub fn depth(block_tree: &BlockTree) -> u32 {
-    if block_tree.children.is_empty() {
-        return 0;
-    }
-
-    let mut max_child_depth = 0;
-
-    for child in block_tree.children.iter() {
-        max_child_depth = std::cmp::max(1 + depth(child), max_child_depth);
+// Returns depth based on the difficulty.
+pub fn difficulty_based_depth(tree: &BlockTree, network: Network) -> u128 {
```

Comment:
nit: can we add a unit test for this method?

Reply:
It is done.